test-data/unit/fine-grained-dataclasses.test

```diff
@@ -0,0 +1,39 @@
+-- Test cases for fine-grained incremental checking of dataclasses
```

Comment:
Maybe make this file more generic, like `fine-grained-plugins.test`? Or just move this test case to the main `fine-grained.test` file.